server/ServerFSM.cpp

```diff
@@ -469,18 +469,20 @@ MPLobby::MPLobby(my_context c) :
 
                 m_lobby_data->m_players.push_back(std::make_pair(player_id, player_setup_data));
             } else if (player_connection->GetClientType() == Networking::CLIENT_TYPE_AI_PLAYER) {
-                PlayerSetupData player_setup_data;
-                player_setup_data.m_player_id =     Networking::INVALID_PLAYER_ID;
-                player_setup_data.m_player_name =   UserString("AI_PLAYER") + "_" + std::to_string(m_ai_next_index++);
-                player_setup_data.m_client_type =   Networking::CLIENT_TYPE_AI_PLAYER;
-                player_setup_data.m_empire_name =   GenerateEmpireName(player_setup_data.m_player_name, m_lobby_data->m_players);
-                player_setup_data.m_empire_color =  GetUnusedEmpireColour(m_lobby_data->m_players);
-                if (m_lobby_data->m_seed != "")
-                    player_setup_data.m_starting_species_name = sm.RandomPlayableSpeciesName();
-                else
-                    player_setup_data.m_starting_species_name = sm.SequentialPlayableSpeciesName(m_ai_next_index);
+                if (m_ai_next_index <= GetOptionsDB().Get<int>("mplobby-max-ai") || GetOptionsDB().Get<int>("mplobby-max-ai") < 0) {
```

Comment:
Seems to work, but there needs to be some indication in the GUI why the game can't be started if due to not having enogh players or AIs.

Reply:
What if send ErrorMessage?

Reply:
Ideally there should be an indication of what's required without having to attempt to start the game. There's currently the text saying individual player names are needed. A greyed out accept button with a tooltip indicating the current requirements would also be good.

Reply:
For grayed out accept button `mplobby-min-humar` should be exposed to the clients.

Reply:
I think you can just add the limits to `MultiplayerLobbyData`.

Depending on the options settings being consistent between clients and server seems problematic...

Reply:
I've just think I can send information about if accept button locked and the lock's cause. So it would be more universal than just send limits.